lib/ansible/modules/system/aix_lvg.py

```diff
@@ -0,0 +1,404 @@
+            rc, _, err = module.run_command(
+                "%s %s %s" % (extendvg_cmd, vg, " ".join(pvs)))
+            if rc == 0:
+                module.exit_json(
+                    changed=True, msg="Volume group %s extended." % vg)
+            else:
+                module.exit_json(
+                    changed=False,
+                    msg="Extending volume group %s has failed." % vg,
+                    rc=rc, err=err)
+
+        return changed
+
+    elif vg_state is None:
+        # Volume group creation
+
+        changed = True
+        if not module.check_mode:
+            mkvg_cmd = module.get_bin_path("mkvg", True)
+            rc, _, err = module.run_command(
+                "%s %s %s %s -y %s %s" % (
+                    mkvg_cmd, vg_opt[vg_type], pp_size, force_opt[force],
+                    vg, ' '.join(pvs)))
+
+            if rc == 0:
+                module.exit_json(
+                    changed=True, msg="Volume group %s created." % vg)
+            else:
+                module.exit_json(
+                    changed=False, msg="Creating volume group %s failed." % vg,
+                    rc=rc, err=err)
+
+        return changed
+
+
+def reduce_vg(module, vg, pvs, vg_validation):
+    vg_state, msg = vg_validation
+
+    if vg_state is False:
+        module.exit_json(
+            changed=False, msg=msg)
+
+    # Define pvs_to_remove (list of physical volumes to be removed).
+    if pvs is None:
+        # Remove VG if pvs are note informed.
+        # Remark: AIX will permit remove only if the VG has not LVs.
+        lsvg_cmd = module.get_bin_path("lsvg", True)
+        rc, current_pvs, err = module.run_command("%s -p %s" % (lsvg_cmd, vg))
+        if rc != 0:
+            module.fail_json(msg="Failing to execute %s command." % lsvg_cmd)
+
+        pvs_to_remove = []
+        for line in current_pvs.splitlines()[2:]:
+            pvs_to_remove.append(line.split()[0])
+
+        reduce_msg = "Volume group %s removed." % vg
+    else:
+        pvs_to_remove = pvs
+        reduce_msg = (
+            "Physical volume(s) %s removed from Volume group %s." % (
+                " ".join(pvs_to_remove), vg))
+
+    # Reduce volume group.
+    if len(pvs_to_remove) > 0:
+        changed = True
+
+        if module.check_mode:
+            reducevg_cmd = module.get_bin_path("reducevg", True)
+            rc, _, err = module.run_command(
+                "%s -df %s %s" % (reducevg_cmd, vg, " ".join(pvs_to_remove)))
+
+            if rc == 0:
+                module.exit_json(changed=True, msg=reduce_msg)
+            else:
+                module.fail_json(
+                    msg="Unable to remove %s." % vg, rc=rc, err=err)
+
+        return changed
+
+
+def state_vg(module, vg, state, vg_validation):
+    vg_state, msg = vg_validation
+
+    if vg_state is None:
+        module.fail_json(msg=msg)
+
+    if state == 'varyon':
+        if vg_state is False:
+            changed = True
```

Comment:
This should be on the function level to avoid multiple definitions of the same variable on multiple places in the same function.

Reply:
Sorry, maybe I didn't get it. 
I tried to follow the example that you gave me previously and I did a mess.
In this case, this block should be like I did below?

```python
    if state == 'varyon':
        if vg_state is False:
            if not module.check_mode:
                varyonvg_cmd = module.get_bin_path("varyonvg", True)
                rc, varyonvg_out, err = module.run_command(
                    "%s %s" % (varyonvg_cmd, vg))
                if rc != 0:
                    module.fail_json(msg="Varyonvg failed.", rc=rc, err=err)
                else:
                    module.exit_json(
                        changed=True,
                        msg="Varyon volume group %s completed." % vg)
            module.exit_json(changed=True)

        else:
            module.exit_json(changed=False, msg=msg)
```

Reply:
Do not call `exit_json()` from inside of this function. The function should return the `changed` value and you should use `exit_json()` at the end of the `main()` function.

Reply:
Indeed. In that case I will return ```changed```and ```msg``` and use ```exit_json()``` in end of ```main()``` and I will perform it for all other functions.

Regarding ```fail_json()``` Can I keep it into the functions, right? If not, I need to create some standard return like:

```python
# type: fail or exit
# changed Fail or True
return type=fail, changed=False, msg=msg, rc=rc, err=err
```

Reply:
You can keep the `fail_json()` in the functions. Only the `exit_json()` use from the `main()` only.